vizro-core/src/vizro/models/_components/form/date_picker.py

```diff
@@ -71,33 +72,15 @@ def build(self):
             output=output,
             inputs=inputs,
         )
-        # clientside callback is required as a workaround when the date-picker is overflowing its parent container
-        # if there is not enough space. Caused by another workaround for this issue:
-        # https://github.com/snehilvj/dash-mantine-components/issues/219
-        clientside_callback(
-            ClientsideFunction(namespace="date_picker", function_name="update_date_picker_position"),
-            output=Output(self.id, "dropdownPosition"),
-            inputs=Input(self.id, "n_clicks"),
-        )
-
-        date_picker_class = dmc.DateRangePicker if self.range else dmc.DatePicker
 
-        # dropdownPosition must be set to bottom-start as a workaround for issue:
-        # https://github.com/snehilvj/dash-mantine-components/issues/219
-        # clearable must be set to False as a workaround for issue:
-        # https://github.com/snehilvj/dash-mantine-components/issues/212
-        # maxDate must be increased by one day, and later on disabledDates must be set as maxDate + 1 day
-        # as a workaround for issue: https://github.com/snehilvj/dash-mantine-components/issues/230
-        date_picker = date_picker_class(
+        date_picker = dmc.DatePickerInput(
             id=self.id,
             minDate=self.min,
             value=init_value,
-            maxDate=self.max + datetime.timedelta(days=1) if self.max else None,
+            maxDate=self.max,
             persistence=True,
             persistence_type="session",
-            dropdownPosition="bottom-start",
-            clearable=False,
-            disabledDates=self.max + datetime.timedelta(days=1) if self.max else None,
+            type="range" if self.range else "default",
             className="datepicker",
             **date_range_picker_kwargs,
```

Comment:
```suggestion
            **date_range_picker_kwargs,
```

Just a comment for myself (no need to push anything yet): If we decide to remove the `range` argument and introduce the `type` argument under a different name instead, this can go and above `date_range_picker_kwargs = {"allowSingleDateInRange": True} if self.range else {}` can be deleted.